File: result/translate/chatgpt/java2python/python/LEXICOGRAPHICAL_CONCATENATION_SUBSTRINGS_STRING.py
def f_gold(s):
    n = len(s)
    sub_count = n * (n + 1) // 2
    arr = [None] * sub_count
    index = 0
    for i in range(n):
        for length in range(1, n - i + 1):
            arr[index] = s[i:i + length]
            index += 1
    arr.sort()
    res = ""
    for i in range(sub_count):
        res += arr[i]
    return res

File: result/translate/starchat/java2cpp/c++/LEXICOGRAPHICAL_CONCATENATION_SUBSTRINGS_STRING.cpp
string f_gold ( string s ) {
    int n = s.length();
    int sub_count = n * (n + 1) / 2;
    vector<string> arr(sub_count);
    int index = 0;
    for (int i = 0; i < n; i++) {
        for (int len = 1; len <= n - i; len++) {
            arr[index++] = s.substr(i, len);
        }
    }
    sort(arr.begin(), arr.end());
    string res;
    for (int i = 0; i < sub_count; i++) {
        res += arr[i];
    }
    return res;
}